### DISCUSSION TITLE
Segmentation error in peacock

### DISCUSSION THREAD

**Discussion by majumdararn** [idaholab/moose#17560] [Q&A Getting Started] [answered]
I am new to moose. I am trying the learn it through the tutorials. Currently, I am trying to run darcy_thermo_mech step 1 simulation.

Unfortunately, Peacock does not work. I tried to open the result file after running the simulation via command line and I tried to open the input file via peacock and run the simulation there and also view the result inside peacock.

I explain the problem here.
If I want to open the result with peacock, I use the following command:
/moose/python/peacock/peacock -r step1_out.i
But, it shows the following error.
Automatic executable search disabled
Segmentation fault (core dumped)

If I want to open the input file with peacock, I use the following command:
/moose/python/peacock/peacock -i step1.i
In this case, peacock opens. For the first time when I ran the simulation within peacock it ran and then when I wanted to view the result. peacock closed suddenly.
The command line message including error is folllowing:
Running command: /home/amajumda/projects_moose/moose/tutorials/darcy_thermo_mech/step01_diffusion/darcy_thermo_mech-opt Outputs/csv=true -i peacock_run_exe_tmp_step1.i
Working directory: /home/amajumda/projects_moose/moose/tutorials/darcy_thermo_mech/step01_diffusion/problems
Finished: Exit code: 0
Segmentation fault (core dumped)

From the second time onwards, peacock opens and the simulation don't run.
The command line error is the following:
Found executable: /home/amajumda/projects_moose/moose/tutorials/darcy_thermo_mech/step01_diffusion/darcy_thermo_mech-opt
Exodus filenames: ['peacock_run_exe_tmp_step1_out.e']
Segmentation fault (core dumped)

**Comment by aeslaughter** [comment, marked as answer]
What operating system are you running?

**Reply by majumdararn** [reply]
Sorry for late reply. I am using linux. Distribution is ubuntu 20.04.02 LTS.

**Reply by aeslaughter** [reply]
Does the input file run from the command line without Peacock? Are you running on a remote machine?

**Reply by majumdararn** [reply]
Yes. Everything is running without Peacock. No, I am not running on a remote machine. I am running on my own laptop.

**Reply by GiudGiud** [reply]
Could you please run this in a debugger and post the backtrace

**Reply by majumdararn** [reply]
Could you please share, how to run in debugger mode? Is it there in the help section of moose?

**Reply by GiudGiud** [reply]
You should be able to adapt these instructions for Moose to peacock.
https://mooseframework.inl.gov/application_development/debugging.html

**Reply by aeslaughter** [reply]
@GiudGiud Unfortunately, there isn't much to do for debugging Peacock, the segmentation faults come from the C++ libraries that are per-compiled and accesses via python APIs. Typically these problems are from VTK or Qt. I debug them by finding a way to reproduce and tracing through the steps that lead to the failure, either using python debugger or print statements. As such, it will be difficult for us to debug this problem. 

Depending how you installed things there might be a change that the you are running with one python (e.g., system python) but the bindings external libraries (VTK, Qt) were built with another (i.e., conda). Given the mention of the Exodus file, I am guessing the problem is coming from VTK.

To see if you can isolate the problem can you try running this script (https://itk.org/Wiki/VTK/Examples/Python/ExodusII/Read). Be sure to change the Exodus filename and variable name to something that exists, perhaps that you have opened in Paraview.

**Reply by majumdararn** [reply]
I tried to run the file, that you shared. It is showing the following error.
2021-04-27 17:43:39.837 (   0.079s) [          3C3F40]  vtkExodusIIReader.cxx:1662   ERR| vtkExodusIIReaderPrivate (0x55d23f1a1130): Could not read nodal result variable pressure.
I did the following:
1) I created a file named vtkChecker.py.
2) copying the content from the link
3) changed the variables filename to 'step1_out.e' and nodal_var to 'pressure'. pressure is the variable that I am also seeing in paraview.
4) I add execute permission for the file 'vtkChecker.py'
5) I run it using ./vtkCheck.py for the first time and then python3 ./vtkChecker.py for the second time. Both show the same error.

Additionally it opens this window.
![Selection_138](https://user-images.githubusercontent.com/69022567/116273111-c30d9880-a781-11eb-91b8-bae6e30766eb.png)

**Reply by aeslaughter** [reply]
"pressure" might not be nodal, but it looks like VTK is working

**Comment by EricPgh** [comment]
I'm in a similar state.  I came to see if there was any new discussions regarding segfaults with peacock.  I just started going through the examples.  I tried to run peacock with both the test app and the ex01 app.  I followed the online documents maybe a month ago for miniconda.  Tonight I did a git clone and continued following the instructions, I double checked my terminal history to verify I didn't skip a step.  I've tried starting peacock through the PATH variable and with an absolute path to moose/python/peacock.  I am local on MacOS big sur 11.2.1.  The ex01 makes and iterates successfully.  I just did a pip install pyqt5 to satisfy peacock dependencies.  It begins opening a window before the very brief segmentation fault message comes back.

**Reply by GiudGiud** [reply]
I don't think Peacock currently supports Big Sur. @aeslaughter will know

**Reply by milljm** [reply]
Unfortunately it does not... There is a PR (a rather old one) in place that is supposed to fix the issue: #17106

**Reply by GiudGiud** [reply]
Peacock should be supported now @EricPgh 
The necessary python packages were updated
